src/main/java/org/jenkinsci/plugins/ghprb/GhprbAdditionalParameterEnvironmentContributor.java

```diff
@@ -3,101 +3,76 @@
-
-        try {
-            putEnvVar(envs, "ghprbTriggerAuthor", cause.getTriggerSender().getName());
-        } catch (Exception e) {}
-        try {
-            putEnvVar(envs, "ghprbTriggerAuthorEmail", cause.getTriggerSender().getEmail());
-        } catch (Exception e) {}
-
-        String triggerAuthorLogin = "";
-        try {
-            triggerAuthorLogin = cause.getTriggerSender().getLogin();
-            if (triggerAuthorLogin == null) {
-                triggerAuthorLogin = "";
-            }
-            putEnvVar(envs, "ghprbTriggerAuthorLogin", triggerAuthorLogin);
-        } catch (Exception e) {}
-
-        setCommitAuthor(cause, envs);
-
-        putEnvVar(envs, "ghprbAuthorRepoGitUrl", cause.getAuthorRepoGitUrl());
-
-        putEnvVar(envs,
-                  "ghprbTriggerAuthorLoginMention",
-                  !triggerAuthorLogin.isEmpty() ? "@" + triggerAuthorLogin : "");
-        putEnvVar(envs, "ghprbPullId", String.valueOf(cause.getPullID()));
-        putEnvVar(envs, "ghprbTargetBranch", cause.getTargetBranch());
-        putEnvVar(envs, "ghprbSourceBranch", cause.getSourceBranch());
-        putEnvVar(envs, "GIT_BRANCH", cause.getSourceBranch());
-
-        // it's possible the GHUser doesn't have an associated email address
-        putEnvVar(envs, "ghprbPullAuthorEmail", cause.getAuthorEmail());
-        putEnvVar(envs, "ghprbPullAuthorLogin", cause.getPullRequestAuthor().getLogin());
-        putEnvVar(envs, "ghprbPullAuthorLoginMention", "@" + cause.getPullRequestAuthor().getLogin());
-
-        putEnvVar(envs, "ghprbPullDescription", escapeText(String.valueOf(cause.getShortDescription())));
-        putEnvVar(envs, "ghprbPullTitle", cause.getTitle());
-        putEnvVar(envs, "ghprbPullLink", String.valueOf(cause.getUrl()));
-        putEnvVar(envs, "ghprbPullLongDescription", escapeText(String.valueOf(cause.getDescription())));
-
-        putEnvVar(envs, "ghprbCommentBody", escapeText(String.valueOf(cause.getCommentBody())));
-
-        putEnvVar(envs, "ghprbGhRepository", escapeText(String.valueOf(cause.getRepositoryName())));
-        putEnvVar(envs, "ghprbCredentialsId", escapeText(String.valueOf(cause.getCredentialsId())));
-
+        ParametersAction pa = run.getAction(ParametersAction.class);
+        for (String param : params) {
+            addParameter(param, pa, envs);
+        }
     }
 
-
-    private void setCommitAuthor(GhprbCause cause,
-                                 EnvVars values) {
-        String authorName = "";
-        String authorEmail = "";
-        if (cause.getCommitAuthor() != null) {
-            authorName = getString(cause.getCommitAuthor().getName(), "");
-            authorEmail = getString(cause.getCommitAuthor().getEmail(), "");
+    private static void addParameter(String key,
+                                     ParametersAction pa,
+                                     EnvVars envs) {
+        ParameterValue pv = pa.getParameter(key);
```

Comment:
Yeah, you're right, only `getParameters` returns a filtered list, I missed that in my review. It should be good then.